cmd/scale.go

```diff
@@ -318,8 +318,9 @@ func (sc *scaleCmd) run(cmd *cobra.Command, args []string) error {
 				continue
 			}
 
-			if sc.windowsPool {
-				_, _, poolindex, err = utils.WindowsVMSSNameParts(*vmss.Name)
+			osPublisher := *vmss.VirtualMachineProfile.StorageProfile.ImageReference.Publisher
+			if strings.EqualFold(osPublisher, "MicrosoftWindowsServer") {
```

Comment:
Maybe we can have a bool var `isWindows` defined above and re-use it here instead of checking again

Reply:
I am not sure about what you mean here, sorry. The check happens only one time: when the vmss corresponding to the agent pool, we want to scale, is found.

Reply:
Ah that makes sense. All good then.

Reply:
My bad, I read the code too fast the first time and didn't realize this was VMSS.